backend/src/main/java/asegroup1/api/models/landregistry/LandRegistryQuerySelect.java

```diff
@@ -1,144 +1,151 @@
-	 * selectable's}.
-	 * 
-	 * @param selectables to initialise the code
-	 */
 	public LandRegistryQuerySelect(Selectable... selectables) {
-		selectableMap = new ArrayList<Selectable>();
-		select(selectables);
+		selectValues = new LinkedHashMap<String, SelectObj>();
+		for (Selectable selectable : selectables) {
+			addSelectValue(selectable, Aggrigation.SAMPLE);
+		}
 	}
 
-	/**
-	 * Initialise the LandRegistryQuerySelect to contain all {@link Selectable
-	 * selectable's}, except the specified exceptions.
-	 * 
-	 * @param selectAll  If true the query will be initialised to contain all
-	 *                   values, if false the query will be initialised to be empty
-	 * @param exceptions Any exceptions to the specified rule.
-	 */
-	public LandRegistryQuerySelect(boolean selectAll, Selectable... exceptions) {
-		selectableMap = new ArrayList<Selectable>();
-		if (selectAll) {
-			selectAll();
-			deselect(exceptions);
+	public void addSelectValue(String referenceName, Aggrigation aggrigation, String aggrigationResult) {
+		selectValues.put(referenceName, new SelectObj(referenceName, aggrigationResult, aggrigation));
+	}
+
+	public void addSelectValue(String referenceName) {
+		addSelectValue(referenceName, Aggrigation.NONE, null);
+	}
+
+	public void addSelectValue(Selectable reference, Aggrigation aggrigation) {
+		addSelectValue(reference.toString(), aggrigation, capitalise(reference.toString()));
+	}
+
+	private String capitalise(String str) {
+		return str.substring(0, 1).toUpperCase() + str.substring(1);
+	}
+
+	public boolean hasValue(String reference) {
+		return selectValues.containsKey(reference);
+	}
+
+	public boolean removeValue(String reference) {
+		if (hasValue(reference)) {
+			selectValues.remove(reference);
+			return true;
 		} else {
-			select(exceptions);
+			return false;
 		}
 	}
 
-	/**
-	 * Select the specified {@link Selectable selectable's}.
-	 * 
-	 * @param selectables to select
-	 */
-	public void select(Selectable... selectables) {
-		for (Selectable selectable : selectables) {
-			if (!selectableMap.contains(selectable)) {
-				selectableMap.add(selectable);
-			}
-		}
+	private SelectObj getSelectObj(String referenceName) {
+		return selectValues.get(referenceName);
 	}
 
-	/**
-	 * Select all {@link Selectable selectable's}.
-	 */
-	public void selectAll() {
-		EnumSet.allOf(Selectable.class).forEach(v -> selectableMap.add(v));
+	public String[] getSelectValues(String reference) {
+		SelectObj obj = getSelectObj(reference);
+		return obj == null ? null : new String[] { obj.referenceName, obj.aggrigation.toString(), obj.aggrigationName };
 	}
 
-	/**
-	 * Deselect the specified {@link Selectable selectable's}.
-	 * 
-	 * @param selectables to deselect
-	 */
-	public void deselect(Selectable... selectables) {
-		for (Selectable selectable : selectables) {
-			selectableMap.remove(selectable);
-		}
+	public String getSelectValueAggrigationName(String reference) {
```

Comment:
should be `getSelectValueAggregationName(`

The same all times the word aggregation appears